internal/nginx/templates/templates_test.go

```diff
@@ -159,3 +167,241 @@ func TestMainForNGINX(t *testing.T) {
 		t.Fatalf("Failed to write template %v", err)
 	}
 }
+
+func enableHealthStatus(cfg nginx.MainConfig) nginx.MainConfig  {
+    cfg.HealthStatus = true
+    return cfg;
+}
+func enableNginxStatus(cfg nginx.MainConfig) nginx.MainConfig  {
+    cfg.NginxStatus = true
+    return cfg;
+}
+func enableNginxStatusAllowCIDRs(cfg nginx.MainConfig, CIDRs []string) nginx.MainConfig  {
+    cfg.NginxStatusAllowCIDRs = CIDRs
+    return cfg;
+}
+
+
+
+func loadNginxMainTemplate(t *testing.T) *template.Template {
+	tmpl, err := template.New(nginxMainTmpl).ParseFiles(nginxMainTmpl)
+	if err != nil {
+		t.Fatalf("Failed to parse template file: %v", err)
+	}
+	return tmpl;
+}
+
+func loadNginxPlusMainTemplate(t *testing.T) *template.Template {
+	tmpl, err := template.New(nginxPlusMainTmpl).ParseFiles(nginxPlusMainTmpl)
+	if err != nil {
+		t.Fatalf("Failed to parse template file: %v", err)
+	}
+	return tmpl;
+}
+
+// nginx plus both - default
+func TestNginxPlusDefaultStatusAllowCIDRs(t *testing.T) {
```

Comment:
thanks for writing these unit tests. However, I don't think these tests (lines 203 - 407) are necessary.  the main template for nginx.conf does not include any complicated logic that and only propagates the values passed to it. 
in your case, you will only have a simple range statement, like below:
```
{{range $value := .NginxStatusAllowCIDRs}}
allow {{$value}};
{{end}}
```
that's why I recommend removing these tests